pydantic_ai_slim/pydantic_ai/models/gemini.py

```diff
@@ -420,22 +425,20 @@ class _GeminiContent(TypedDict):
     parts: list[_GeminiPartUnion]
 
 
-def _content_user_prompt(m: UserPromptPart) -> _GeminiContent:
-    return _GeminiContent(role='user', parts=[_GeminiTextPart(text=m.content)])
+def _part_user_prompt(m: UserPromptPart) -> _GeminiPartUnion:
+    return _GeminiTextPart(text=m.content)
 
 
-def _content_tool_return(m: ToolReturnPart) -> _GeminiContent:
-    f_response = _response_part_from_response(m.tool_name, m.model_response_object())
-    return _GeminiContent(role='user', parts=[f_response])
+def _part_tool_return(m: ToolReturnPart) -> _GeminiPartUnion:
+    return _response_part_from_response(m.tool_name, m.model_response_object())
```

Comment:
Since these are only called once and are very simple, I don't think we need functions, just inline the logic.

Reply:
inlined in 5a56dc4